### DISCUSSION TITLE
Interpretation of loss values for YourTTS training

### DISCUSSION THREAD

**Discussion by fColangelo** [coqui-ai/TTS#1306] [General Q&A] [answered]
Hi everyone and thanks for the awesome work.
I am looking for some guidance on how to interpret the loss curves in the below picture, as well as some more general tips on how to check the progress of the training
I am currently training a 4 language model starting from the YourTTS experiment 3 checkpoint, adding the italian language from the M-AILABS dataset. 
In the plots, it would seem that the optimum is around 60K iterations, after that most of the curves start to overfit, except avg_loss_mel and avg_loss_spk_encoder. Would it be correct to select the 60K model as best? What value contribute the most to the quality of the model in your experience?
![losses_VITS](https://user-images.githubusercontent.com/22350076/156031647-4443656c-9b08-4eff-89cf-e32e18847ef0.png)
.
Thanks in advance!

**Comment by Edresson** [comment, marked as answer]
Hello @fColangelo, 

You can choose the best checkpoint based on the average loss of the generator (avg_loss_gen + avg_loss_kl + avg_loss_mel + avg_loss_spk_encoder+avg_loss_feat + avg_loss_duration) in this case the avg_loss_0 represents the sum (it can be changed in the next TTS versions, so if you are reading it after new releases keep attention on this). 

However, in many cases, it is not so accurate. If you check the avg_loss_spk_encoder you will see that the similarity of the speaker's voices is increasing and the Mel spectrogram is turning near to the Ground truth.  But the Duration is turning worst.  In the end, the better way to select the best checkpoint is to listen to samples for each checkpoint. The target loss can change depending in what is your priority for the TTS model.  If you want a better voice similarity you need to look at avg_loss_spk_encoder for example. However, in many cases, the use of Speaker Consistency Loss (avg_loss_spk_encoder) can produce the worst quality than the model trained without this extra loss.

**Comment by fColangelo** [comment]
Hi @Edresson, thanks for your answer!
These insights are invaluable.
